packages/openapi/src/helpers.ts

```diff
@@ -164,3 +171,113 @@ export function isReadonlyProperty(program: Program, property: ModelProperty) {
   // readonly: true, but using separate schemas.
   return visibility?.length === 1 && visibility[0] === "read";
 }
+
+/**
+ * Determines if a OpenAPIExtensionKey is start with `x-`.
+ */
+export function isOpenAPIExtensionKey(key: string): key is ExtensionKey {
+  return key.startsWith("x-");
+}
+
+/**
+ * Validate that the given string is a valid URL.
+ * @param program  Program
+ * @param target Diagnostic target for any diagnostics that are reported
+ * @param url The URL to validate
+ * @param propertyName The name of the property that the URL is associated with
+ * @returns true if the URL is valid, false otherwise
+ */
+export function validateIsUri(
+  program: Program,
+  target: DiagnosticTarget,
+  url: string,
+  propertyName: string,
+): boolean {
+  try {
+    // Attempt to create a URL object from the given string. If
+    // successful, the URL is valid.
+    new URL(url);
+    return true;
+  } catch {
+    // If the URL is invalid, report a diagnostic with the given
+    // target, property name and value.
+    reportDiagnostic(program, {
+      code: "not-url",
+      target: target,
+      format: { property: propertyName, value: url },
+    });
+    return false;
+  }
+}
+
+/**
+ * Validate the AdditionalInfo model against a reference.
+ *
+ * This function checks that the properties of the given AdditionalInfo object
+ * are a subset of the properties defined in the AdditionalInfo model.
+ *
+ * @param program - The TypeSpec Program instance
+ * @param target - Diagnostic target for reporting any diagnostics
+ * @param typespecType - The AdditionalInfo object to validate
+ * @param reference - The reference string to resolve the model
+ * @returns true if the AdditionalInfo object is valid, false otherwise
+ */
+export function validateAdditionalInfoModel(
+  program: Program,
+  target: DiagnosticTarget,
+  typespecType: TypeSpecValue,
+  reference: string,
+): boolean {
+  // Resolve the reference to get the corresponding model
+  const propertyModel = program.resolveTypeReference(reference)[0]! as Model;
+
+  // Check if typespecType is an object and propertyModel is defined
+  if (typeof typespecType === "object" && propertyModel) {
+    // Validate that the properties of typespecType do not exceed those in propertyModel
+    const diagnostics = checkNoAdditionalProperties(typespecType, target, propertyModel);
+    program.reportDiagnostics(diagnostics);
+    // Return false if any diagnostics were reported, indicating a validation failure
+    if (diagnostics.length > 0) {
+      return false;
+    }
+  }
+
+  // Return true if validation is successful
+  return true;
+}
+
+/**
+ * Check Additional Properties
+ */
+function checkNoAdditionalProperties(
+  typespecType: Type,
+  target: DiagnosticTarget,
+  source: Model,
+): Diagnostic[] {
+  const diagnostics: Diagnostic[] = [];
+  compilerAssert(typespecType.kind === "Model", "Expected type to be a Model.");
```

Comment:
compiler asssert should never be used to validate, it means you are confident that this will never happen(and if it does its a bug and reported as such to the user) which here is not the case as the typespecType can be any type change to `: Model`